### ISSUE TITLE
Collaboration not working inside file but works in terminal.

### ISSUE THREAD

**Issue by vanyle** [jupyterlab/jupyter-collaboration#253]
## Description

While I can see cursors and selections of other people, the file is not saved automatically and I cannot see what other users are typing. I does not work for notebooks or plain text files.
When using a terminal however, collaboration works properly.

![image](https://github.com/jupyterlab/jupyter-collaboration/assets/9964043/a59c1bbf-1f98-47fe-8355-eeefbcfadada)

## Reproduce

I'm using the following `docker-compose.yml` file:

```yaml
services:
  jupyter:
    image:  jupyter/pyspark-notebook:latest
    restart: unless-stopped
    ports:
      - "8888:8888"
    user: root
    working_dir: /home/vanyle
    environment:
      - NB_USER=vanyle
      - CHOWN_HOME=yes
    command: "start-notebook.py --ServerApp.token='' --ServerApp.password=''"
    volumes:
      - ./models:/home/vanyle/work
```
When the docker is started, I manually install `jupyter_collaboration` and I restart the docker.

## Expected behavior

I would expect *jupyter-collaboration* to work. When a character is typed on one device, it appears on the other.

## Context

- Operating System: The issue occurs on all OS I tested, including Windows 11 and MacOS (meaning the docker is executed inside MacOS / inside Windows).
- Browser and version: The issue arises on all browser I tested, including Firefox, Edge and Chrome.
- JupyterLab version: 4.1.4
- Jupyter collaboration version: 2.0.3

I have the following plugins installed:

![image](https://github.com/jupyterlab/jupyter-collaboration/assets/9964043/51a6b62f-98c8-4bdb-abc5-f5f150669f64)

**Comment by davidbrochart** [comment]
What do you mean by "works in terminal"?

**Comment by davidbrochart** [comment]
I yanked pycrdt v0.8.13 on PyPI, marked it as broken on conda-forge, and released v0.8.14 with a fix.

**Comment by vanyle** [comment]
It works! Thank you! I spend like 2 hours tweaking the configuration of my docker (as I'm using a custom docker with a lot of extensions) and nothing was working.

Just out of curiosity, what was the issue with `pycrdt` ?

**Comment by davidbrochart** [comment]
In `pycrdt` v0.8.13 the Rust backend [Yrs](https://github.com/y-crdt/y-crdt/tree/main/yrs) was updated to v0.18, which changed the way observer subscriptions are handled. Now instead of getting a subscription ID, a subscription object is returned, and unobserving is done by dropping the object, so in Python a reference to the subscription has to be kept, otherwise it gets garbage-collected and it is dropped on the Rust side (see https://github.com/y-crdt/y-crdt/issues/398).
I had done that for shared types in https://github.com/jupyter-server/pycrdt/pull/85 but I forgot to do it for docs, and that's now fixed in https://github.com/jupyter-server/pycrdt/pull/87.
Basically all this means is that the changes to a document in JupterLab was not propagated to other users.
